src/hera/workflows/script.py

```diff
@@ -606,27 +617,81 @@ def _output_annotations_used(source: Callable) -> bool:
 
 FuncIns = ParamSpec("FuncIns")  # For input types of given func to script decorator
 FuncR = TypeVar("FuncR")  # For return type of given func to script decorator
+FuncRCov = TypeVar("FuncRCov", covariant=True)
 
 ScriptIns = ParamSpec("ScriptIns")  # For input attributes of Script class
-StepIns = ParamSpec("StepIns")  # # For input attributes of Step class
-TaskIns = ParamSpec("TaskIns")  # # For input attributes of Task class
 
 
-# Pass actual classes of Script, Step and Task to bind inputs to the ParamSpecs above
+class _ScriptDecoratedFunction(Generic[FuncIns, FuncRCov], Protocol):
+    """Type assigned to functions decorated with @script."""
+
+    # Note: name is required for Step + Task, but is automatically inferred from
+    #       the name of the decorated function for @script.
```

Comment:
thanks for answering! much clear. other things looks good.
how about adding other note comment here saying like following parameters should match with the parameter of step or task? I think this is the first place to look when the mypy raises the errors about script decorated objects.